tests/Feature/DemoTest.php

```diff
@@ -11,14 +21,14 @@ class DemoTest extends TestCase
 	 * Check that the demo page is not available
 	 * if not enabled in the advanced config.
 	 */
-	public function testDemo0()
+	public function testDemo0(): void
 	{
 		// save initial value
 		$init_config_value = Configs::get_value('gen_demo_js');
 
 		// set to 0
 		Configs::set('gen_demo_js', '0');
-		$this->assertEquals(Configs::get_value('gen_demo_js'), '0');
+		static::assertEquals('0', Configs::get_value('gen_demo_js'));
```

Comment:
Is there a particular reason for the flip? (`0==i` instead of `i==0`)

Reply:
Yes, making the Linter happy and getting correct error messages.

The order of arguments is `expected`, `actual`. Hence, the desired argument must come first. This will also give you a reasonable error message in case the test fails, i.e. something like that "<foo> is expected to equal <expected>, but <value of foo> received". If the arguments are swapped, the error message becomes nonsense.

My Linter knows that. Basically it only triggers for comparison with constant values like here and if the constant value is in the 2nd argument, then the Linter complains.

Reply:
Ok. Unintuitive, but this is PHP.

Reply:
Totally. I would prefer the other way round, too. I can only guess that it is this way, because there are specialized methods like `assertEuqalsNull`, `assertEqualsZero`, etc., where the expected value is encoded into the name of the method and thus also comes first.